api/src/main/java/com/codeforcommunity/dto/report/ReportGenericListResponse.java

```diff
@@ -0,0 +1,34 @@
+package com.codeforcommunity.dto.report;
+
+import java.util.List;
+
+public class ReportGenericListResponse {
+
+  private int count;
+  private List<ReportGeneric> reports;
+
+  public ReportGenericListResponse(List<ReportGeneric> reports) {
+    if (reports == null) {
+      throw new IllegalArgumentException("Given `null` list of reports");
+    }
+
+    this.reports = reports;
```

Comment:
I am assuming the reports are immutable because they are dto's so not converting and returning a new array list will be necessary right?

Reply:
Good thinking, but I think its okay in this case because it's just being used as a DTO, yes